lib/WasmReader/WasmSignature.cpp

```diff
@@ -206,24 +240,54 @@ uint32 WasmSignature::WriteSignatureToString(_Out_writes_(maxlen) char16* out, u
 {
     AssertOrFailFast(out != nullptr);
     uint32 numwritten = 0;
-    numwritten += _snwprintf_s(out+numwritten, maxlen-numwritten, _TRUNCATE, _u("("));
-    for (Js::ArgSlot i = 0; i < this->GetParamCount(); i++)
+#define WRITE_BUF(msg, ...) numwritten += _snwprintf_s(out+numwritten, maxlen-numwritten, _TRUNCATE, msg, ##__VA_ARGS__);
+    WRITE_BUF(_u("("));
+    for (Js::ArgSlot i = 0; i < GetParamCount(); i++)
     {
         if (i != 0)
         {
-            numwritten += _snwprintf_s(out+numwritten, maxlen-numwritten, _TRUNCATE, _u(", "));
+            WRITE_BUF(_u(", "));
         }
-        numwritten += _snwprintf_s(out+numwritten, maxlen-numwritten, _TRUNCATE, _u("%ls"), WasmTypes::GetTypeName(this->GetParam(i)));
+        WRITE_BUF(_u("%ls"), WasmTypes::GetTypeName(this->GetParam(i)));
     }
     if (numwritten >= maxlen-12) {
         // null out the last 12 characters so we can properly end it 
         for (int i = 1; i <= 12; i++) {
             *(out + maxlen - i) = 0;
         }
-        numwritten -= 12;
-        numwritten += _snwprintf_s(out + numwritten, maxlen - numwritten, _TRUNCATE, _u("..."));
+        if (numwritten < 12)
+        {
+            numwritten = 0;
+        }
+        else
+        {
+            numwritten -= 12;
+        }
+
+        WRITE_BUF(_u("..."));
+    }
+    if (GetResultCount() == 0)
+    {
+        WRITE_BUF(_u(")->void"));
+    }
+    else if (GetResultCount() == 1)
+    {
+        WRITE_BUF(_u(")->%ls"), WasmTypes::GetTypeName(this->GetResult(0)));
+    }
+    else
+    {
+        WRITE_BUF(_u(")->("));
+        for (uint32 i = 0; i < GetResultCount(); i++)
```

Comment:
nit: You may want to add handling for maxlen here and report ")->(…)" or something like that if there's not enough space.

Reply:
done